model/model.go

```diff
@@ -82,6 +83,12 @@ func NewPackage(pkgpath string, useExtTypes bool) *Package {
 	if semver.IsValid(name) {
 		// if the name is a valid semver, use the parent's name
 		name = path.Base(path.Dir(pkgpath))
+		// TODO: there are false positives, like k8s.io/apimachinery/pkg/apis/meta/v1,
```

Comment:
I'm not sure this actually matters; does it?

Reply:
The package name will need to be changed manually in the `stub.go` file, otherwise it won't compile.

Reply:
Does the stub.go not use `Name` when generating imports?

Reply:
> Does the stub.go not use Name when generating imports?

It does.

The issue is that `name = path.Base(path.Dir(pkgpath))` is not always valid.

Reply:
I don't see the problem since the name is sanitized afterward anyway. On that note, I think the check for keyword should be in the sanitize function as well.

Reply:
The ideal way to get the package name would be from `pacgaes.Load`.

If the package name is wrong, `sanitize` won't fix that.

I'll try to solve that in another PR.

Reply:
The package name is set in import by the generator using this name, so that multiple packages with the same name can be imported. This should be fine.

Reply:
It is completely fine, unless it's not a semver but the actual package name.

It's an edge case anyway, so not really worth the effort for now.